pom.xml

```diff
@@ -58,6 +60,12 @@
       <artifactId>junit-jupiter</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.platform</groupId>
+      <artifactId>org.eclipse.osgi</artifactId>
+      <version>3.17.100</version>
+      <scope>test</scope>
+    </dependency>
```

Comment:
This is needed for the resolve goal. I moved into the dependenciesManagement section to ensure it is not actually on the unit test execution path.

Reply:
I wanted to move it directly in the plugin dependencies, but I realized that the build succeeds even without that dependency declaration. Can it be that the plugins are including already the needed dependencies?

Reply:
Plugins and dependencies are separate. Adding the dependency under the plugin would not help since the plugin's execution does not need the dependency. But the resolve operation will look in the project's dependencies and dependencyManagement section for suitable resources. So this dependency must appear in either of those places. Using dependencyManagement is cleaner since the dependency is not put on the test case execution classpath.

Reply:
What I meant is if I remove the dependency completely I still get correct executions:
* Build is successful with `-runsystempackages: org.assertj.core.internal `
* Build fails without `-runsystempackages: org.assertj.core.internal`

I'm probably lacking some fundamental knowledge of the `org.eclipse.osgi` Maven artifact. Is there some documentation where I could educate myself?

Reply:
`org.eclipse.osgi` is an OSGi framework implementation. It is not necessary for the resolve goal to function. But once #70 is fixed, we should add some actual OSGi integration testing like https://github.com/assertj/assertj-core/commit/c9807b9c9a300758ee262ea5026acd3be68781cf to the build. At that time, we will need an OSGi framework implementation like `org.eclipse.osgi`. 

So you can remove `org.eclipse.osgi` now, but it would need to be added back later when actual OSGi integration testing is added.

Reply:
Clear and thank you, @bjhargrave.

Reply:
I'm going to remove that declaration as the way we want to go is most likely assertj/assertj-core#2424.